### DISCUSSION TITLE
How to enable showing internal repositories for everyone

### DISCUSSION THREAD

**Discussion by temamagic** [woodpecker-ci/woodpecker#4299] [Ask a Question]
I added a repository, made it internal (and even public), but only I can see it, other users can only see it if they open it via a direct link. How can I make it so that they can see shared repositories simply in the list?

**Comment by qwerty287** [comment]
They need write access in the  forge to the repo

**Reply by temamagic** [reply]
Could you please clarify in more detail? We use GitHub, we have an organization and everyone has access to it

As an admin, I also do not see shared repositories in the general list (/repos), only in the admin panel

What rights should be given and how?

**Reply by qwerty287** [reply]
So you have write access to the repos *on github*? It works for us over here. I can see all the repos at the @woodpecker-ci org I have write access to in my personal repo list.

**Reply by temamagic** [reply]
Why we talking about repos in github, i'm talking about repos in woodpecker ci ui

I administer ci, the developers added their repositories, I made them "internal" and neither they nor I see each other in the general tab of repositories, but we see each other via a direct link (or I via the admin panel)

**Reply by anbraten** [reply]
You have to be a member of a github project to have it popup in the list in woodpeckers as woodpecker is using the permissions from your forge. The list only shows "your" repos. 

Image being on a larger public Woodpecker instance with a lot of repos all of them (like all public repos from github) would be shown otherwise to a user, although they only care about their own ones.

**Reply by temamagic** [reply]
But there is no point in this, because the repository is available to me via a direct link and is open inside the CI, I can go through all the IDs inside /repos/1+... and get access to them

Why can't I get a button in the interface where "Available" to me will be?

Then it is worth renaming "Repositories" to "My repositories"

**Reply by qwerty287** [reply]
> Why can't I get a button in the interface where "Available" to me will be?

Actually, there are open discussions about this to have some kind of "explore" screen. That would probably solve it for you.

**Reply by qwerty287** [reply]
This is listed in https://github.com/woodpecker-ci/woodpecker/discussions/2162#discussion-5498655

I'm closing this in favor of this discussion. If you still don't think that will solve it, please comment again.
